partner_event/wizard/res_partner_register_event.py

```diff
@@ -31,9 +31,16 @@ def _prepare_registration(self, partner):
     def button_register(self):
         vals_list = []
         Registration = self.env["event.registration"]
-        for partner in self.env["res.partner"].browse(
-            self.env.context.get("active_ids", [])
-        ):
+        active_ids = self.env.context.get("active_ids", [])
+        if self.env.context.get("active_model", "") == "event.registration":
+            partners = (
+                self.env["event.registration"]
+                .browse(active_ids)
+                .mapped("attendee_partner_id")
```

Comment:
What if not all registration have `attendee_partner_id`? That can happen if partner auto-create is not checked.

Reply:
I just refined the method to consider that possibility